buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectGenerate.java

```diff
@@ -0,0 +1,403 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.buildtools.parser;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.lang.StringBuilder;
+import java.lang.IllegalStateException;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+import java.util.LinkedList;
+import java.util.LinkedHashMap;
+import java.util.Queue;
+import java.util.regex.Matcher;
+import java.util.regex.MatchResult;
+import java.util.regex.Pattern;
+
+public class DialectGenerate {
+
+  private final String typeAndName = "\\w+\\s+\\w+";
+  private final String splitDelims = "(\\s|\n|\"|//|/\\*|\\*/|'|\\}|\\{)";
+
+  private final Pattern tokenizerPattern = Pattern.compile("((?<="
+      + splitDelims + ")|(?=" + splitDelims + "))");
+  private final Pattern declarationPattern =
+    Pattern.compile("(" + typeAndName + "\\s*\\(\\s*(" + typeAndName + "\\s*(\\,\\s*"
+        + typeAndName + "\\s*)*)?\\)\\s*\\:\n?)");
+  private final Pattern namePattern = Pattern.compile("\\w+");
+
+  private final File dialectDirectory;
+  private final File rootDirectory;
+  private final String outputFile;
+
+  public DialectGenerate(File dialectDirectory, File rootDirectory, String outputFile) {
+    this.dialectDirectory = dialectDirectory;
+    this.rootDirectory = rootDirectory;
+    this.outputFile = outputFile;
+  }
+
+  /**
+   * Extracts functions and generates parserImpls.ftl for the given dialect.
+   */
+  public void run() {
+    Map<String, String> functions = extractFunctions();
+    for (Map.Entry<String, String> entry : functions.entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+      System.out.println(key + "=" + value + "\n");
```

Comment:
Comment says that it generates parserImpls.ftl, but code seems to suggest that all it does is print to console.

Reply:
It will be in the next PR.

Reply:
I'll add a comment explaining this

Reply:
Agree with @lisayin that we should have the PR make sense standalone as much as possible. Can we have the comment accurately reflect that we only print for now. And in the follow up PR change it to reflect the new behavior?

Alternatively, you could leave the comment as is and write a "TODO(apochapsky): <description_of_what_to_do>" comment to make it clear to a reader/reviewer that there is an intention to change the logic.